### ISSUE TITLE
E-mail falsely marked as Spam with reason ' Failed Captcha “Javascript”: “” ' on multi-page form

### ISSUE THREAD

**Issue by Antimated** [verbb/formie#681]
**Description**
Since I've updated Formie from 1.4.26 to 1.5.1 all my e-mails are getting marked as spam with the spam reason being `Failed Captcha “Javascript”: “”`.  I do have javascript enabled in my browser. Whilst filling in the form I don't get any issues.

Also when I try to send a notification from a submission that is marked as spam I get the "A server occurred" error flash message.

**Steps to reproduce**
1. In my case I enabled the following captchas: Duplicate, Honeypot and Javascript
2. Create a multi-page form
3. Go through the steps and finish your form
4. Check if the submission was put in the "spam" folder.

### Form settings
- Multi-page form: Yes
- Submission Method: Ajax
- Client-side Validation: Yes
- Custom Form Templates: Yes although only the submit button was changed slightly for visual purposes only

**Additional info**
- Plugin version: 1.5.1
- Craft version: 3.7.25.1
- Multi-site: No

**Comment by engram-design** [comment]
The only reason you should get an empty string in the spam reason (`Javascript: “”`) is when the payload param is not a string. This is to protect invalid data being set to this param `__JSCHK_xxx` which is persisted over a session. 

So a few things:
- Can you verify if the payload being send to Formie contains this param? Use your Web Developer > Network looks to look at the request being send when submitting a form
- Does your site use Vue/React?
- Is your site using statically cached templates, or cache template tags?

In all other cases, where the param doesn't match, not provided, or expired a spam reason should be included in the spam submission.

**Comment by Antimated** [comment]
@engram-design 
- I looked at the payload and it seems to be including 2 `__JSCHK__xxx`'s:
`beesknees:
__JSCHK_1314441: 61b1bf031d3c9
__JSCHK_1314441: 61b1bf031d3c9
`
- My site uses `Vue` but our site is ran as a [async module (link to formie docs)](https://verbb.io/craft-plugins/formie/docs/developers/javascript-api#async-modules):
`            if (window.Formie) {
                window.Formie.initForms();
            }`
- The site uses cache tags but **_we made sure to not wrap any forms in them_** so this should not be an issue.

Furthermore single page forms do work.

**Comment by Antimated** [comment]
> The only reason you should get an empty string in the spam reason (`Javascript: “”`) is when the payload param is not a string. This is to protect invalid data being set to this param `__JSCHK_xxx` which is persisted over a session.
> 
> So a few things:
> 
> * Can you verify if the payload being send to Formie contains this param? Use your Web Developer > Network looks to look at the request being send when submitting a form
> * Does your site use Vue/React?
> * Is your site using statically cached templates, or cache template tags?
> 
> In all other cases, where the param doesn't match, not provided, or expired a spam reason should be included in the spam submission.

- I looked at the payload and it seems to be including 2 `__JSCHK__xxx`'s:
`beesknees:
__JSCHK_1314441: 61b1bf031d3c9
__JSCHK_1314441: 61b1bf031d3c9
`
- My site uses `Vue` but our site is ran as a [async module (link to formie docs)](https://verbb.io/craft-plugins/formie/docs/developers/javascript-api#async-modules):
So in the created method we added this:
`            if (window.Formie) {
                window.Formie.initForms();
            }`
- The site uses cache tags but **_we made sure to not wrap any forms in them_** so this should not be an issue.

Furthermore single page forms do work.

(if you want the full payload let me know)

**Comment by engram-design** [comment]
Yep, you're right, it's multi-page ajax forms that seem to trip this up. I'll investigate!

**Comment by engram-design** [comment]
Fixed for the next release. To get this fix early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-3 as 1.5.1",
  "...": "..."
}
```

Then run `composer update`.

**Comment by engram-design** [comment]
Fixed in [1.5.2](https://github.com/verbb/formie/releases/tag/1.5.2)
